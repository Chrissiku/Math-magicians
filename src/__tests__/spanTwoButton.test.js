/* eslint-disable react/jsx-props-no-spreading */
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import SpanTwo from '../components/component_mocks/DigitButton';

const props = { name: 'Chris', e: jest.fn() };
describe('Test span two button', () => {
  test('Test Span two Button component', () => {
    const TREE = renderer
      .create(<SpanTwo e={props.e} name={props.name} />)
      .toJSON();

    expect(TREE).toMatchSnapshot();
  });
  it('Test button 0 click event', () => {
    render(<SpanTwo {...props} />);
    fireEvent.click(screen.getByText(props.name));
    expect(props.e).toHaveBeenCalled();
    expect(props.e).toHaveBeenCalledTimes(1);
  });
});
