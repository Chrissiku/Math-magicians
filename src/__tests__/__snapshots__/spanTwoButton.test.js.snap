// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test span two button Test Span two Button component 1`] = `
<button
  onClick={[MockFunction]}
  type="button"
>
  Chris
</button>
`;
